build-image-gallery-astro-cloudflare.mdx

```diff
@@ -0,0 +1,456 @@
+
+  // Set the relevant message for the user
+  response.message = 'Published profile succesfully. Redirecting you to upload your first photograph...'
+}
+---
+
+<!-- Render conditional states using the onboarded flag -->
+{
+  response.form &&
+    (response.created ? (
+      <p class="rounded bg-green-100 px-3 py-1">{response.message}</p>
+    ) : (
+      <p class="rounded bg-red-100 px-3 py-1">{response.message}</p>
+    ))
+}
+
+<!-- Profile Form -->
+
+<!-- Redirect user to the next location -->
+<script define:vars={{ success: response.created, location: response.redirect }}>
+  if (success) {
+    setTimeout(() => {
+      window.location.href = location
+    }, 1000)
+  }
+</script>
+```
+
+## Using Xata query
+
+The user profile page (`src/pages/[profile]/index.astro`) leverages the Xata Client to dynamically fetch and display all the photographs (not paginated) for a specific user profile. To engage users visually as soon as they open the gallery, we use the stored `imageHash` value as the background of the images (to be loaded). To prevent CLS, we use the stored `width` and `height` values to inform the browser of the expected dimensions of the images.
+
+```tsx
+---
+// File: src/pages/[profile]/index.astro
+
+// Import the Xata Client created by Xata CLI in src/xata.ts
+import { getXataClient } from '@/xata'
+
+import Layout from '@/layouts/Layout.astro'
+
+// Get the profile slug from url path
+const { profile } = Astro.params
+
+// Fetch the Xata instance
+const xata = getXataClient()
+
+// Get all the photographs related to the profile
+const profilePhotographs = await xata.db.photographs
+														// Filter the results to the specific profile
+														.filter({ 'profile-slug': profile })
+														// Get all the photographs
+														.getAll()
+---
+
+<Layout className="flex flex-col">
+  <div class="columns-1 gap-0 md:columns-2 lg:columns-3">
+    {
+      profilePhotographs.map(
+        ({ width: photoW, height: photoH, name: photoName, image: photoImageURL, tagline: photoTagline, imageHash: photoImageHash }, _) => (
+					{/* Destructure the width and height to prevent CLS */}
+          <img
+            width={photoW}
+            height={photoH}
+            alt={photoName}
+            src={photoImageURL}
+						{/* Do not lazy load the first image that's loaded into the DOM */}
+            loading={_ === 0 ? 'eager' : 'lazy'}
+            class="transform bg-cover bg-center bg-no-repeat will-change-auto"
+						{/* Create a blur effect with the imageHash stored */}
+            style={`background-image: url(${photoImageHash}); transform: translate3d(0px, 0px, 0px);`}
+          />
+        ),
+      )
+    }
+  </div>
+</Layout>
+```
+
+![(Image gallery - presenting blur images initially)](/images/build-image-gallery/image15.png)
+
+(Image gallery - presenting blur images initially)
+
+![(Image gallery - after image loads complete)](/images/build-image-gallery/image15.png)
```

Comment:
This is the same image as the one on line 414

So:
Image gallery - presenting blur images initially
Image gallery - after image loads complete

are the same. What did we mean to show here?